src/Sarif.Driver/Sdk/AnalyzeCommandBase.cs

```diff
@@ -101,6 +99,59 @@ private void Analyze(TOptions analyzeOptions, AggregatingLogger logger)
             }
         }
 
+        protected virtual void ValidateOptions(TContext context, TOptions analyzeOptions)
+        {
+            bool succeeded = true;
+
+            succeeded &= ValidateFile(context, analyzeOptions.OutputFilePath, shouldExist: null);
+            succeeded &= ValidateFile(context, analyzeOptions.ConfigurationFilePath, shouldExist: true);
+            succeeded &= ValidateFiles(context, analyzeOptions.PlugInFilePaths, shouldExist: true);
+
+            if (!succeeded)
+            {
+                ThrowExitApplicationException(context, ExitReason.InvalidCommandLineOption);
+            }
+        }
+
+        private bool ValidateFiles(TContext context, IEnumerable<string> filePaths, bool shouldExist)
+        {
+            bool succeeded = true;
+
+            foreach (string filePath in filePaths)
+            {
+                succeeded &= ValidateFile(context, filePath, shouldExist);
+            }
+
+            return succeeded;
+        }
+
+        private bool ValidateFile(TContext context, string filePath, bool? shouldExist)
+        {
+            Exception exception = null;
+
+            try
+            {
+                bool fileExists = File.Exists(filePath);
+
+                if (shouldExist == null || shouldExist.Value)
```

Comment:
I get that you put the `null` test after the call to `File.Exists` so you could see if you got an exception attempting to access the directory path. But the rest of the condition is wrong because it doesn't use the result of the call to `File.Exists`. You would return `true` here even if `fileExists` were `false`.